### ISSUE TITLE
PKSimCreatePopulation: returned parameters

### ISSUE THREAD

**Issue by Yuri05** [Open-Systems-Pharmacology/Matlab-Toolbox-Core#12]
In the list of parameters returned by `PKSimCreatePopulation `we have **_Volume_** and **_Blood flow rate_** of mucosa segments and liver zones, e.g. 
```
'Organism|SmallIntestine|Mucosa|Duodenum|Volume'
'Organism|SmallIntestine|Mucosa|Duodenum|Blood flow rate'
'Organism|Liver|Periportal|Volume'
'Organism|Liver|Periportal|Blood flow rate'
```

Any idea, why? In mucosa segments and liver zones both parameters are defined **by formulas**.

And also 
`'Organism|Fat|Fraction intracellular'` (also defined by formula)

<details><summary>Parameter list returned by PKSimCreatePopulation</summary>

```
'Organism|Weight'
'Organism|BMI'
'Organism|BSA'
'Organism|Gestational age'
'Organism|Age'
'Organism|Ontogeny factor (alpha1-acid glycoprotein)'
'Organism|Ontogeny factor (albumin)'
'Organism|Height'
'Organism|Hematocrit'
'Organism|MeanBW'
'Organism|MeanHeight'
'Organism|VenousBlood|Volume'
'Organism|ArterialBlood|Volume'
'Organism|Bone|Blood flow rate'
'Organism|Bone|Specific blood flow rate'
'Organism|Bone|Volume'
'Organism|Brain|Blood flow rate'
'Organism|Brain|Volume'
'Organism|Brain|Specific blood flow rate'
'Organism|Fat|Fraction intracellular'
'Organism|Fat|Blood flow rate'
'Organism|Fat|Volume'
'Organism|Fat|Vf (neutral lipid)-PT'
'Organism|Fat|Vf (lipid)'
'Organism|Fat|Vf (phospholipid)-PT'
'Organism|Fat|Fraction interstitial'
'Organism|Fat|Vf (water)'
'Organism|Fat|Vf (water)-PT'
'Organism|Fat|Vf (neutral lipid)-RR'
'Organism|Fat|Vf (neutral lipid)-WS'
'Organism|Fat|Vf (neutral phospholipid)-RR'
'Organism|Fat|Vf (neutral phospholipid, plasma)-WS'
'Organism|Fat|Vf (extracellular water)-RR'
'Organism|Fat|Vf (intracellular water)-RR'
'Organism|Fat|Vf (water)-WS'
'Organism|Fat|Specific blood flow rate'
'Organism|Gonads|Blood flow rate'
'Organism|Gonads|Volume'
'Organism|Gonads|Specific blood flow rate'
'Organism|Heart|Blood flow rate'
'Organism|Heart|Volume'
'Organism|Heart|Specific blood flow rate'
'Organism|Kidney|GFR (specific)'
'Organism|Kidney|Blood flow rate'
'Organism|Kidney|Volume'
'Organism|Kidney|Maximal decreasing rate factor'
'Organism|Kidney|TM50 for GFR'
'Organism|Kidney|Age of aging onset'
'Organism|Kidney|GFRmat'
'Organism|Kidney|Aging half-time'
'Organism|Kidney|Hill coefficient for GFR'
'Organism|Kidney|Hill coefficient for aging GFR'
'Organism|Kidney|Specific blood flow rate'
'Organism|Lumen|Effective surface area variability factor'
'Organism|Lumen|Stomach|Distal radius'
'Organism|Lumen|Stomach|GET_beta (Weibull function) variability factor'
'Organism|Lumen|Stomach|Gastric emptying time'
'Organism|Lumen|Stomach|GET_alpha (Weibull function) variability factor'
'Organism|Lumen|Stomach|Length'
'Organism|Lumen|Stomach|Proximal radius'
'Organism|Lumen|Duodenum|Effective surface area enhancement factor'
'Organism|Lumen|UpperJejunum|Effective surface area enhancement factor'
'Organism|Lumen|LowerJejunum|Effective surface area enhancement factor'
'Organism|Lumen|UpperIleum|Effective surface area enhancement factor'
'Organism|Lumen|LowerIleum|Effective surface area enhancement factor'
'Organism|Lumen|Caecum|Effective surface area enhancement factor'
'Organism|Lumen|ColonAscendens|Effective surface area enhancement factor'
'Organism|Lumen|ColonTransversum|Effective surface area enhancement factor'
'Organism|Lumen|ColonDescendens|Effective surface area enhancement factor'
'Organism|Lumen|ColonSigmoid|Effective surface area enhancement factor'
'Organism|Lumen|Rectum|Effective surface area enhancement factor'
'Organism|Stomach|Blood flow rate'
'Organism|Stomach|Volume'
'Organism|Stomach|Specific blood flow rate'
'Organism|SmallIntestine|Blood flow rate'
'Organism|SmallIntestine|Small intestinal transit time'
'Organism|SmallIntestine|Volume'
'Organism|SmallIntestine|Specific blood flow rate'
'Organism|SmallIntestine|Mucosa|Duodenum|Volume'
'Organism|SmallIntestine|Mucosa|Duodenum|Blood flow rate'
'Organism|SmallIntestine|Mucosa|UpperJejunum|Volume'
'Organism|SmallIntestine|Mucosa|UpperJejunum|Blood flow rate'
'Organism|SmallIntestine|Mucosa|LowerJejunum|Volume'
'Organism|SmallIntestine|Mucosa|LowerJejunum|Blood flow rate'
'Organism|SmallIntestine|Mucosa|UpperIleum|Volume'
'Organism|SmallIntestine|Mucosa|UpperIleum|Blood flow rate'
'Organism|SmallIntestine|Mucosa|LowerIleum|Volume'
'Organism|SmallIntestine|Mucosa|LowerIleum|Blood flow rate'
'Organism|LargeIntestine|Blood flow rate'
'Organism|LargeIntestine|Large intestinal transit time'
'Organism|LargeIntestine|Volume'
'Organism|LargeIntestine|Specific blood flow rate'
'Organism|LargeIntestine|Mucosa|Caecum|Volume'
'Organism|LargeIntestine|Mucosa|Caecum|Blood flow rate'
'Organism|LargeIntestine|Mucosa|ColonAscendens|Volume'
'Organism|LargeIntestine|Mucosa|ColonAscendens|Blood flow rate'
'Organism|LargeIntestine|Mucosa|ColonTransversum|Volume'
'Organism|LargeIntestine|Mucosa|ColonTransversum|Blood flow rate'
'Organism|LargeIntestine|Mucosa|ColonDescendens|Volume'
'Organism|LargeIntestine|Mucosa|ColonDescendens|Blood flow rate'
'Organism|LargeIntestine|Mucosa|ColonSigmoid|Volume'
'Organism|LargeIntestine|Mucosa|ColonSigmoid|Blood flow rate'
'Organism|LargeIntestine|Mucosa|Rectum|Volume'
'Organism|LargeIntestine|Mucosa|Rectum|Blood flow rate'
'Organism|Liver|Blood flow rate'
'Organism|Liver|Volume'
'Organism|Liver|Specific blood flow rate'
'Organism|Liver|Periportal|Volume'
'Organism|Liver|Periportal|Blood flow rate'
'Organism|Liver|Pericentral|Volume'
'Organism|Liver|Pericentral|Blood flow rate'
'Organism|Lung|Blood flow rate'
'Organism|Lung|Volume'
'Organism|Lung|Fraction vascular'
'Organism|Muscle|Fraction intracellular'
'Organism|Muscle|Blood flow rate'
'Organism|Muscle|Volume'
'Organism|Muscle|Vf (neutral lipid)-PT'
'Organism|Muscle|Vf (lipid)'
'Organism|Muscle|Vf (protein)'
'Organism|Muscle|Vf (phospholipid)-PT'
'Organism|Muscle|Vf (water)-PT'
'Organism|Muscle|Vf (water)'
'Organism|Muscle|Fraction interstitial'
'Organism|Muscle|Vf (neutral lipid)-WS'
'Organism|Muscle|Vf (neutral lipid)-RR'
'Organism|Muscle|Vf (neutral phospholipid, plasma)-WS'
'Organism|Muscle|Vf (neutral phospholipid)-RR'
'Organism|Muscle|Vf (extracellular water)-RR'
'Organism|Muscle|Vf (protein)-WS'
'Organism|Muscle|Vf (intracellular water)-RR'
'Organism|Muscle|Vf (water)-WS'
'Organism|Muscle|Specific blood flow rate'
'Organism|Pancreas|Blood flow rate'
'Organism|Pancreas|Volume'
'Organism|Pancreas|Specific blood flow rate'
'Organism|PortalVein|Blood flow rate'
'Organism|PortalVein|Volume'
'Organism|Skin|Blood flow rate'
'Organism|Skin|Volume'
'Organism|Skin|Specific blood flow rate'
'Organism|Spleen|Blood flow rate'
'Organism|Spleen|Volume'
'Organism|Spleen|Specific blood flow rate'
'Organism|Saliva|Volume'
```
</details>

**Comment by PavelBal** [comment]
`'Organism|Weight'` is also defined by a formula, and it is returned, too. I complained about it years ago :)

**Comment by Yuri05** [comment]
Some parameters which you can see in PK-Sim under "Characteristics of Individual" (Weight/Height/BSA/BMI) are returned on purpose, so that in Matlab the user can get this info directly without having to calculate it manually.

**Comment by msevestre** [comment]
>Any idea, why? In mucosa segments and liver zones both parameters are defined by formulas.

The implementation simply returns all parameters that have the flag `IsChangedByCreateIndividual` set to `true`

This are all parameter named
```
  Constants.Parameters.VOLUME,
            AGE,
            GESTATIONAL_AGE,
            HEIGHT,
            WEIGHT,
            MEAN_WEIGHT,
            MEAN_HEIGHT,
            BMI,
            BSA
```
as well as

```
  /// <summary>
         ///    List of parameters that are indirectly varied by create individual (such as Blood Flow, GFR_spec etc.) and that
         ///    should be displayed in a distribution plot
         /// </summary>
         public static readonly IReadOnlyList<string> DerivedCreatedIndividualParameters = new[]
            {
               BLOOD_FLOW,
               GFR_SPEC,
               VOLUME_PLASMA,
            }
            .Union(OntogenyFactors)
            .Union(AllPlasmaProteinOntogenyFactors).ToList();
```

and then special cases
```
    if (CoreConstants.Parameters.VolumeFractionWaterParameters.Contains(Name) && isInFatOrMuscle())
               return true;

            //formula parameter indirectly changed by create individual 
            if (string.Equals(CoreConstants.Parameters.FRACTION_INTRACELLULAR, Name) && isInFatOrMuscle())
               return true;

            if (CoreConstants.Parameters.VolumeFractionLipidsParameters.Contains(Name) && isInFat())
               return true;

            if (CoreConstants.Parameters.VolumeFractionProteinsParameters.Contains(Name) && isInMuscle())
               return true;
```

The reason being that those parmaeters should be displayed as distribution and as such their "individual sepcific values" are also saved.

It would be easy to filter out some parameter for matlab export...it just needs to be specified

## To summarize:
The fact that a parameter is a formula or not has no bearing in the decision to export a parameter or not.
